addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/ahamodel/TemperatureModel.java

```diff
@@ -30,24 +31,30 @@
     private BigDecimal offset;
 
     public BigDecimal getCelsius() {
-        return celsius != null ? celsius.multiply(TEMP_FACTOR) : BigDecimal.ZERO;
+        return celsius;
     }
 
     public void setCelsius(BigDecimal celsius) {
         this.celsius = celsius;
     }
 
     public BigDecimal getOffset() {
-        return offset != null ? offset.multiply(TEMP_FACTOR) : BigDecimal.ZERO;
+        return offset;
     }
 
     public void setOffset(BigDecimal offset) {
         this.offset = offset;
     }
 
+    public static BigDecimal toCelsius(BigDecimal fritzValue) {
+        if (fritzValue == null) {
+            return BigDecimal.ZERO;
+        }
+        return TEMP_FACTOR.multiply(fritzValue);
```

Comment:
1) Maybe also rename the TEMP_FACTOR to indicate to what conversion factor it refers
2) did you check all occurrences where the existing methods were called?